jep/0000/README.adoc

```diff
@@ -0,0 +1,345 @@
+the Jenkins project
+
+== Specification
+
+=== Overview
+
+We'll use a short story to illustrate what Autorelease will look like when a plugin maintainer 
+named Karl decides that he wants to perform continuous delivery of his (fictional) Karl-Plugin.
+
+To do so, Karl will:
+
+1. Verify support. Karl's plugin is already in the 
+link:https://github.com/jenkinsci[jenkinsci Organization on GitHub], and is 
+using `buildPlugin` when built on ci.jenkins.io.
+
+2. Choose a branch from which to release. Karl decides he wants to release from the `master` branch, 
+as that is the simplest.
+
+3. Settle on a version numbering scheme. Becuase it's in fairly wide use across the Jenkins 
+ecosystem, Karl settles on link:https://semver.org/[semantic versioning].
+
+4. Add a file to the top directory of his repository called `Autorelease.yaml`. 
+
+5. A webhook is configured on Karl's repository on GitHub to point to `{the-webhook-receiver}`
+
+Once ready to create a new release of his plugin, Karl merges some changes to the `master` branch. 
+At which point, the following takes place under the covers: 
+
+1. A webhook is sent to `{the-webhook-receiver}` by GitHub
+
+2. The receiver forwards this hook to `${trusted-ci-server}`, which is protected by a VPN
+
+3. `${trusted-ci-server}` uses the enhanced version of `buildPlugin` to validate the correctness of 
+Karl's `Autorelease.yaml` file, and the consistency of Karl's chosen version numbering.
+
+4. Once `Autorelease.yaml` has been validated, a plugin build is performed on `${trusted-ci-server}` 
+as one would expect.
+
+5. After the build is successful, `${trusted-ci-server}` handles deployment of the built plugin to 
+Artifactory and the Jenkins Update Center.
+
+=== Detailed Setup
+
+==== Autorelease.yaml
+
+A YAML markerfile, stored in the plugin's own repository. Absence of this markerfile will 
+be functionally equivalent to the plugin maintainer opting out of Autorelease. 
+
+In the following example, we'll stick with plugin developer Karl, and his fictional 
+plugin `karl-plugin`. Karl's `Autorelease.yaml` file looks like this:
+
+```
+plugin: karl
+kind: plugin
+metadata:
+  autoreleaseEnabled: true
+  triggeringBranch: master
+  versionNumberFields: 3
+```
+
+In the above example `Autorelease.yaml`, we can see the following:
+
+`plugin: karl`:: 
+This is the name of the plugin, and it matches the repository 
+name on GitHub.
+`kind`:: 
+The type of project which is being set up for autorelease. In our first iteration of 
+autorelease, this will only be set to `plugin`. The field is here should the Jenkins project 
+want to expand the use of autorelease, such as for building core, or components.
+`metadata`:: 
+Contains the following fields:
+** `autoreleaseEnabled`: A boolean value, indicating whether or not this plugin is participating 
+in autorelease. Allows the maintainer to turn Autorelease on or off, without deleting the file 
```

Comment:
IMHO drop this field. We can comfortably delete files, because we have version control.

Reply:
Fair enough, I'll strike it.